### PR TITLE
HPCC-13807 Give warning when too many files

### PR DESCRIPTION
**Pull request by miguelvazq**
Whenever there is over 100,000 files we will issue a warning to the user to set a filter.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 1]
```diff
@@ -350,6 +351,22 @@ define([
...
+        },
+
+        showFilesWarningDialog: function (numberOfFiles) {
+            var context = this;
+            var fileWarningDialog = new Dialog({
+                title: this.i18n.TooManyFiles,
+                content: this.i18n.TheReturnedResults + " " + numberOfFiles + " " + this.i18n.RepresentsASubset,
+                style: "width: 300px"
+            });
+            fileWarningDialog.show();
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: We would normally popup a toaster message at this point?

**Comment by richardkchapman** [comment]
@miguelvazq Back to you

File: esp/src/eclwatch/DFUQueryWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/DFUQueryWidget.js]: I can implement that way as well.

**Comment by miguelvazq** [comment]
@richardkchapman  @GordonSmith Implemented

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 2]
```diff
@@ -350,6 +351,18 @@ define([
             topic.subscribe("hpcc/dfu_wu_completed", function (topic) {
                 context.refreshGrid();
             });
+
+            WsDfu.DFUQuery({
+                request: {}
+            }).then(function (response) {
+                if (response.DFUQueryResponse.Warning) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: Minor:  Should have a lang.exists to test if it exists

Comment (miguelvazq) [inline esp/src/eclwatch/DFUQueryWidget.js]: Good catch. Will send fix here now.

**Comment by miguelvazq** [comment]
@GordonSmith updated.

**Comment by GordonSmith** [comment]
@richardkchapman Looks good.

**Pull request merged**